File: new/cpp/dynamic_array.cpp
class DynamicArray {
private:
    int *arr; // arr is pointer to list of int
    int size;
    int capacity;

public:

    DynamicArray(int capacity) : capacity(capacity), size(0) {
        arr = new int[capacity];
    }

    int get(int i) {
        // assuming i is valid
        return arr[i];
    }

    void set(int i, int n) {
        // assuming i is valid
        arr[i] = n;
    }

    void pushback(int n) {
        if (size == capacity) {
            resize();
        }

        arr[size] = n;
        size += 1;
    }

    int popback() {
        size -= 1;
        return arr[size];
    }

    void resize() {
        capacity = capacity * 2;
        int *newArr = new int[capacity];
        for (int i = 0; i < size; i++) {
            newArr[i] = arr[i];
        }

        delete[] arr;
        arr = newArr;
    }

    int getSize() {
        return size;
    }

    int getCapacity() {
        return capacity;
    }
};

File: new/python/dynamic_array.py
class DynamicArray:
    def __init__(self, capacity: int):
        self.size = 0
        self.capacity = capacity
        self.arr = [0] * self.capacity

    def get(self, i: int) -> int:
        # assuming i is valid
        return self.arr[i]

    def set(self, i: int, n: int) -> None:
        # assuming i is valid
        self.arr[i] = n

    def pushback(self, n: int) -> None:
        if self.size == self.capacity:
            self.resize()

        self.size += 1
        self.arr[self.size - 1] = n

    def popback(self) -> int:
        # we don't need to delete it
        self.size -= 1
        return self.arr[self.size]

    def resize(self) -> None:
        self.capacity = self.capacity * 2
        newArr = [0] * self.capacity
        for i in range(len(self.arr)):
            newArr[i] = self.arr[i]
        self.arr = newArr

    def getSize(self) -> int:
        return self.size

    def getCapacity(self) -> int:
        return self.capacity
